tailcfg/tailcfg.go

```diff
@@ -571,12 +571,31 @@ func (h *Hostinfo) Equal(h2 *Hostinfo) bool {
 	if h == nil && h2 == nil {
 		return true
 	}
-	if (h == nil) != (h2 == nil) {
+	if h == nil || h2 == nil {
 		return false
 	}
 	return reflect.DeepEqual(h, h2)
 }
 
+// BasicallyEqual reports whether h and h2 are equal other than the
+// NetInfo DERP latency timing. (see NetInfo.BasicallyEqual).
+func (h *Hostinfo) BasicallyEqual(h2 *Hostinfo) bool {
+	if h == nil && h2 == nil {
+		return true
+	}
+	if h == nil || h2 == nil {
+		return false
+	}
+	a := *h
+	b := *h2
+	if !a.NetInfo.BasicallyEqual(b.NetInfo) {
+		return false
+	}
+	a.NetInfo = nil
+	b.NetInfo = nil
+	return a.Equal(&b)
```

Comment:
I think we need to filter out Services. (And that means our future plans of adding potentially interesting services to Services should be rethought.)

There's also a lot of machines out there that bounce their Hostname around. Not sure what to do about that one. Definitely not startup critical, but also not really "basically equal"